sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,668 @@
+        """
+        X_meta = []
+        for est_idx, preds in enumerate(predictions):
+            # case where the the estimator returned a 1D array
+            if preds.ndim == 1:
+                X_meta.append(preds.reshape(-1, 1))
+            else:
+                if self.stack_method_[est_idx] == 'predict_proba':
+                    # Remove the first column when using probabilities. It is
+                    # collinear with others.
+                    X_meta.append(preds[:, 1:])
+                else:
+                    X_meta.append(preds)
+        return np.concatenate(X_meta, axis=1)
+
+    @staticmethod
+    def _method_name(name, estimator, method):
+        if estimator == 'drop':
+            return None
+        if method == 'auto':
+            if getattr(estimator, 'predict_proba', None):
+                return 'predict_proba'
+            elif getattr(estimator, 'decision_function', None):
+                return 'decision_function'
+            else:
+                return 'predict'
+        else:
+            if not hasattr(estimator, method):
+                raise ValueError('Underlying estimator {} does not implement '
+                                 'the method {}.'.format(name, method))
+            return method
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like of shape (n_samples,) or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+        names, estimators_ = self._validate_estimators()
+        self._validate_final_estimator()
+
+        has_estimator = False
+        for est in estimators_:
+            if est != 'drop':
+                has_estimator = True
+
+        if not has_estimator:
+            raise ValueError(
+                "All estimators are dropped. At least one is required "
+                "to be an estimator."
+            )
+
+        stack_method = [self.stack_method] * len(estimators_)
+
+        # Fit the base estimators on the whole training data. Those
+        # base estimators will be used in transform, predict, and
+        # predict_proba. They are exposed publicly.
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(
+            delayed(_parallel_fit_estimator)(clone(est), X, y, sample_weight)
+            for est in estimators_ if est != 'drop'
+        )
+
+        self.named_estimators_ = Bunch()
+        est_fitted_idx = 0
```

Comment:
maybe use enumerate?

Reply:
nop because I want to not count when an estimator is dropped.